sql/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/Functions.scala

```diff
@@ -202,19 +206,17 @@ case class ST_Transform(inputExpressions: Seq[Expression])
 
   override def eval(input: InternalRow): Any = {
     val geometry = inputExpressions(0).toGeometry(input)
-    val sourceCRS = inputExpressions(1).asString(input)
-    val targetCRS = inputExpressions(2).asString(input)
+    val sourceCRSString = inputExpressions(1).asString(input)
+    val targetCRSString = inputExpressions(2).asString(input)
 
-    (geometry, sourceCRS, targetCRS) match {
-      case (geometry: Geometry, sourceCRS: String, targetCRS: String) =>
-        val lenient = if (inputExpressions.length == 4) {
-          inputExpressions(3).eval(input).asInstanceOf[Boolean]
-        } else {
-          false
-        }
-        Functions.transform(geometry, sourceCRS, targetCRS, lenient).toGenericArrayData
-      case (_, _, _) => null
+    val lenient = if (inputExpressions.length == 4) {
+      inputExpressions(3).eval(input).asInstanceOf[Boolean]
+    } else {
+      false
     }
+
+    (Functions transform(geometry, sourceCRSString, targetCRSString, lenient)).toGenericArrayData
```

Comment:
Nit: just use dot notation like everywhere else
```suggestion
    Functions.transform(geometry, sourceCRSString, targetCRSString, lenient).toGenericArrayData
```